conda_libmamba_solver/solver.py

```diff
@@ -444,23 +447,54 @@ def _specs_to_tasks_add(self, in_state: SolverInputState, out_state: SolverOutpu
             + list(in_state.history.keys())
             + list(in_state.aggressive_updates.keys())
         )
+
+        # Fast-track python version changes
+        # ## When the Python version changes, this implies all packages depending on
+        # ## python will be reinstalled too. This can mean that we'll have to try for every
+        # ## installed package to result in a conflict before we get to actually solve everything
+        # ## A workaround is to let all non-noarch python-depending specs to "float" by marking
+        # ## them as a conflict preemptively
+        python_version_might_change = False
+        installed_python = in_state.installed.get("python")
+        to_be_installed_python = out_state.specs.get("python")
+        if installed_python is not None and to_be_installed_python is not None:
+            python_version_might_change = not to_be_installed_python.match(installed_python)
+
         tasks = defaultdict(list)
         for name, spec in out_state.specs.items():
             if name.startswith("__"):
                 continue
             self._check_spec_compat(spec)
             spec_str = str(spec)
+            installed = in_state.installed.get(name)
             key = "INSTALL", api.SOLVER_INSTALL
+
+            # Fast-track Python version changes: mark non-noarch Python-depending packages as
+            #  conflicting (see `python_version_might_change` definition above for more details)
```

Comment:
```suggestion
            # conflicting (see `python_version_might_change` definition above for more details)
```

Reply:
@chenghlee Hey, do you know if this change will actually be moot by our plans to deprecate of noarch for interpreter based packages?

Reply:
There are plans to deprecate the `noarch: python` functionality? I wasn't aware of this. Is there a CEP covering the rationale? It seems like a pretty big change.